web/frontend/src/components/application/AppDetails.tsx

```diff
@@ -28,6 +30,18 @@ const AppDetails: React.FC = () => {
     fetchData()
   }, [])
 
+  const getHealth = async (): Promise<void> => {
+    try {
+      if (runtimePid) {
+        await getServiceHealth(runtimePid)
+        setError(undefined)
+      }
+    } catch (error) {
+      setError(error)
+    }
+  }
+  useInterval(() => { getHealth() }, INTERVAL_HEALTH)
```

Comment:
I have a rough feeling this will be instantiated a lot as this component can be rendered multiple times. Can you verify?

Reply:
I have not noticed any issue when using this approach. Anyway, the code could be simplified (and we can avoid that potential re-render problem) by checking the service health before getting the metrics. You can have a look at e1010cca8c5f0d616678947bd007994eb1e9f1f8